### PR TITLE
Issue #3462: Prevent fatal error for missing image.

### LINKED ISSUE #3462: Saving an Image field with an image that has undeterminable dimensions throws PDOException.
**Describe the bug**
Saving an Image field with an image that has undeterminable dimensions throws a fatal error. You'd think a missing file isn't that big of a deal, right? 

```
EntityStorageException: SQLSTATE[HY000]: General error: 1366 Incorrect integer value: '' for column 'field_image_width' at row 1 in NodeStorageController->save() (line 320 of backdrop/core/modules/node/node.entity.inc).
```

**Steps To Reproduce**
To reproduce the behavior:
1. Upload an image onto an article, save the node.
2. Delete the file from disk so it no longer exists in the location Backdrop is tracking.
3. Edit/Save the same page again.
4. Fatal error.

**Actual behavior**
Fatal error.

**Expected behavior**
No fatal error.

**Additional information**
- [Related d.o issue](https://www.drupal.org/project/drupal/issues/1330952)

---
PR https://github.com/backdrop/backdrop/pull/2459

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/3462

### CODE REVIEW

File: core/modules/image/image.field.inc [hunk 1]
```diff
@@ -292,13 +292,18 @@ function image_field_presave($entity_type, $entity, $field, $instance, $langcode
...
   foreach ($items as &$item) {
-    if (!isset($item['width']) || !isset($item['height'])) {
+    if (empty($item['width']) || empty($item['height'])) {
       $info = image_get_info(file_load($item['fid'])->uri);
 
       if (is_array($info)) {
         $item['width'] = $info['width'];
         $item['height'] = $info['height'];
       }
+      else {
```
Comment (herbdool) [inline core/modules/image/image.field.inc]: You could put the variable defaulting to NULL before the if statement. That way it's clear they'll always be NULL unless width and height are available.

Comment (jenlampton) [reply core/modules/image/image.field.inc]: good idea. done!

Comment (herbdool) [reply core/modules/image/image.field.inc]: Better. My suggestion also applies to the first part of this patch as well. The setting of NULL can be above the if statement.

Comment (jenlampton) [reply core/modules/image/image.field.inc]: of course :) done!

**Comment by jenlampton** [comment]
looks like one of our random failures

**Comment by jenlampton** [comment]
no longer an issue.

**Pull request closed without merging**